dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -747,6 +748,33 @@ public User updateLoggedInUserMetadata(@ApiParam(hidden = true)@Parameter(hidden
         return dbuser;
     }
 
+    @PUT
+    @Timed
+    @UnitOfWork
+    @RolesAllowed({"admin", "curator"})
+    @Path("/user/{userId}/privileges")
+    @Operation(operationId = "addUserPrivileges", description = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiOperation(value = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class)
+    public User setUserPrivilege(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User authUser,
+                                 @ApiParam(value = "User ID", required = true) @PathParam("userId") Long userID,
+                                 @ApiParam(value = "Set privilege for a user", required = true) PrivilegeRequest privilegeRequest) {
+        User user = userDAO.findById(userID);
+        if (user == null) {
+            throw new CustomWebApplicationException("User not found", HttpStatus.SC_NOT_FOUND);
+        }
+
+        if (privilegeRequest.isAdmin() != user.getIsAdmin() && !authUser.getIsAdmin()) {
```

Comment:
Not sure I'm reading correctly
It seems possible to simplify, is this something akin to : 
only admins can create admins, curators and admins can create curators?

Reply:
Right, only admins can create/remove admins, but both admins and curators can create/remove curators

Reply:
Agreed, it is a little complex and a little hard to read.

I suggest just not allowing a user to modify their own privileges, period. Add that as a test after line 764. Then  the logic is not buried in another else if.

Reply:
Added the logic to prevent a user to modify their own privileges after line 764.